editor/script_create_dialog.cpp

```diff
@@ -304,16 +281,8 @@ String ScriptCreateDialog::_validate_path(const String &p_path, bool p_file_must
 	return "";
 }
 
-String ScriptCreateDialog::_get_class_name() const {
-	if (has_named_classes) {
-		return class_name->get_text();
-	} else {
-		return ProjectSettings::get_singleton()->localize_path(file_path->get_text()).get_file().get_basename();
-	}
-}
-
 void ScriptCreateDialog::_class_name_changed(const String &p_name) {
-	is_class_name_valid = _validate_class(class_name->get_text());
+	is_class_name_valid = class_name->get_text().is_empty() || class_name->get_text().is_valid_identifier();
```

Comment:
Maybe we should check if the class has a name already used by a native or a global class. Then, we could output a custom error message about that.